### DISCUSSION TITLE
How to use tanstak queryClient (or queries/mutations) in Vue custom directives?

### DISCUSSION THREAD

**Discussion by viktorkrepak** [TanStack/query#5769] [Q&A]
Hey! I've been using Tanstack with my Vue3 application and it works awesome, but I've got stuck with one tricky issue. 
I wanted to create a `hasRole` directive to check a user roles and based on the input value (array of acceptable roles) make a decision whether I can render a component for this user or no. 
I'm fetching the user data using a query and It's saved in my the tanstack cache and I don't want to duplicate this data to Pinia store, but when I try to use Tanstack queryClient inside of my custom directive I get this error `Error: vue-query hooks can only be used inside setup() function.`. So to temporary fix that I've passed the user data using directive input parameters, but It looks like a dirty solution. Can smb help me with the issue? Can I somehow use the queryClient in Vue custom directives?

**Comment by Mini-ghost** [comment]
@viktorkrepak

Perhaps this PR https://github.com/TanStack/query/pull/5703 could solve your issue, but currently, there are still some limitations, and it needs to wait for Vue's PR https://github.com/vuejs/core/pull/8801 to be completed before it can be implemented.

**Reply by viktorkrepak** [reply]
thanks a lot for your reply!
